package.json

```diff
@@ -65,27 +67,31 @@
     "markdown-it": "13.0.1",
     "node-fetch": "^3.3.1",
     "recheck": "^4.4.5",
-    "rollup": "^3.25.1",
     "semantic-release": "^21.0.5",
     "titleize": "^3.0.0",
+    "ts-expect": "^1.3.0",
+    "ts-node": "^10.9.1",
+    "tsup": "^6.7.0",
+    "typescript": "5.0.4",
     "uglify-js": "^3.17.4",
     "vuln-regex-detector": "^1.3.0"
   },
   "scripts": {
-    "test": "jasmine --config=jasmine.json",
+    "test": "cross-env NODE_OPTIONS=--loader=ts-node/esm jasmine --config=jasmine.json",
     "test:all": "npm test && npm run test:lint",
     "test:unit": "npm test -- test/unit/**/*-spec.js",
     "test:specs": "npm test -- test/specs/**/*-spec.js",
     "test:lint": "eslint .",
     "test:redos": "node test/recheck.js > vuln.js",
     "test:update": "node test/update-specs.js",
     "rules": "node test/rules.js",
-    "bench": "npm run rollup && node test/bench.js",
+    "bench": "npm run build && node test/bench.js",
     "lint": "eslint --fix .",
+    "type-check": "tsc",
```

Comment:
This isn't really `type-check`, its running `tsc` which will compile.

If you only want type checking without emitting js output, it should be `tsc --noEmit`